### PR TITLE
[ALLUXIO-3077] Add source of different configuration properties to track

### PR DESCRIPTION
**Pull request by apc999**
https://alluxio.atlassian.net/browse/ALLUXIO-3077

- Change in `getConf` CLI, a new `--source` option is added to show the source of a configuration in the following two cases:

```bash
$ bin/alluxio getConf --source
alluxio.conf.dir=/Users/binfan/Dropbox/Work/alluxio/alluxio/conf (SYSTEM_PROPERTY)
alluxio.debug=true (SITE_PROPERTY: /Users/binfan/Dropbox/Work/alluxio/alluxio/conf/alluxio-site.properties)
alluxio.extensions.dir=${alluxio.home}/extensions (DEFAULT)
...
$ bin/alluxio getConf --source alluxio.debug
SITE_PROPERTY: /Users/binfan/Dropbox/Work/alluxio/alluxio/conf/alluxio-site.properties
```

- Change in Configuration tab of WebUI: add a new column to show the source of each configuration property.

screen shot 2017-12-14 at 2 48 25 pm

### CODE REVIEW

File: core/common/src/main/java/alluxio/Configuration.java [hunk 1]
```diff
@@ -163,39 +174,24 @@ private static Properties createDefaultProps() {
...
         String key = entry.getKey().toString().trim();
         String value = entry.getValue().toString().trim();
         if (PropertyKey.isValid(key)) {
+          PropertyKey propertyKey = PropertyKey.fromString(key);
+          if (source == Source.SITE_PROPERTY && propertyKey.isIgnoredSiteProperty()) {
+            LOG.warn("{} is not accepted in alluxio-site.properties, "
+                    + "and must be specified as a JVM property. "
+                    + "If no JVM property is present, Alluxio will use default value '{}'.",
+                key, propertyKey.getDefaultValue());
+            continue;
```
Comment (calvinjia) [inline core/common/src/main/java/alluxio/Configuration.java]: We may want to crash here.

Comment (apc999) [reply core/common/src/main/java/alluxio/Configuration.java]: move this logic to validation and will crash there to be consistent.

**Comment by apc999** [comment]
@calvinjia PTAL

File: core/server/master/src/main/java/alluxio/web/WebInterfaceConfigurationServlet.java [hunk 2]
```diff
@@ -16,11 +16,8 @@
 import alluxio.master.file.FileSystemMaster;
 
 import com.google.common.collect.Sets;
-
-import org.apache.commons.lang3.tuple.ImmutablePair;
-import org.apache.commons.lang3.tuple.Pair;
-
-import com.google.common.collect.Sets;
+import org.apache.commons.lang3.tuple.ImmutableTriple;
+import org.apache.commons.lang3.tuple.Triple;
```
Comment (calvinjia) [inline core/server/master/src/main/java/alluxio/web/WebInterfaceConfigurationServlet.java]: Related to this PR: we should probably come up with something like a delimiter and just pass a String

File: core/common/src/main/java/alluxio/Configuration.java [hunk 3]
```diff
@@ -68,12 +68,24 @@
 public final class Configuration {
   private static final Logger LOG = LoggerFactory.getLogger(Configuration.class);
 
+  /** The source of a configuration property. */
+  public enum Source {
+    UNKNOWN,
```
Comment (yupeng9) [inline core/common/src/main/java/alluxio/Configuration.java]: describe the priority order? Or use alphabetical order

Comment (apc999) [reply core/common/src/main/java/alluxio/Configuration.java]: done

**Pull request merged**